File: micropython/rfid/message.py
from communication_protocol.communication_protocol import DeviceMessage
from communication_protocol.message_event import MessageEvent
from communication_protocol.message_type import MessageType


def on_read_request(mac: str, uid: str) -> DeviceMessage:
    return DeviceMessage(
        MessageEvent.ON_READ,
        MessageType.REQUEST,
        mac,
        {"uid": uid},
    )


def on_click_request(mac: str) -> DeviceMessage:
    return DeviceMessage(
        MessageEvent.ON_CLICK,
        MessageType.REQUEST,
        mac,
        {},
    )


def add_tag_response(mac: str, uid: str, name: str, message_id) -> DeviceMessage:
    return DeviceMessage(
        MessageEvent.ADD_TAG,
        MessageType.RESPONSE,
        mac,
        {"uid": uid, "name": name},
        message_id=message_id,
    )

File: arduino/gate_esp8266_12E/src/CommunicationProtocol/message.cpp
#include "message.h"

DeviceMessage on_read_request(String mac, String uid) {
  JsonDocument payload;
  payload["uid"] = uid;
  return DeviceMessage(String(millis()), "on_read", "request", mac, payload);
}

DeviceMessage on_click_request(String mac) {
  JsonDocument payload;
  return DeviceMessage(String(millis()), "on_click", "request", mac, payload);
}

DeviceMessage on_hold_request(String mac) {
  JsonDocument payload;
  return DeviceMessage(String(millis()), "on_hold", "request", mac, payload);
}

DeviceMessage add_tag_response(DeviceMessage message, String uid) {
  JsonDocument payload;
  payload["uid"] = uid;
  payload["name"] = message.payload["name"].as<String>();
  return DeviceMessage(message.message_id, "add_tag", "response", message.device_id, payload);
}
DeviceMessage accept_response(DeviceMessage message) {
  JsonDocument payload;
  payload["status"] = "accept";
  return DeviceMessage(message.message_id, message.message_event, "response", message.device_id, payload);
}